### PR TITLE
security fix: Migrate from log4j to reload4j

### PR DESCRIPTION
**Pull request by mkandaswamy**
[KCI-1900](https://confluentinc.atlassian.net/browse/KCI-1900): Migrate from log4j to reload4j to fix CVEs

### CODE REVIEW

**Review by jnh5y — APPROVED** [review]

**Comment by mkandaswamy** [comment]
@jnh5y / @niteshmor: thanks for reviewing! I've cherry-picked and opened PRs to fix this issue on 
* 5.5.x - https://github.com/confluentinc/kafka-streams-examples/pull/430
* 6.0.x - https://github.com/confluentinc/kafka-streams-examples/pull/431
* 6.1.x - https://github.com/confluentinc/kafka-streams-examples/pull/432
* 6.2.x - https://github.com/confluentinc/kafka-streams-examples/pull/433
* 7.0.x - https://github.com/confluentinc/kafka-streams-examples/pull/434
* 7.1.x - https://github.com/confluentinc/kafka-streams-examples/pull/435
It would be great if one of you can approve it for all the above branches too(7.2.x to master has been already fixed in Q2). 

(Once I get your approval, I can merge all of them together and will run a pint merge with ours strategy.)

**Pull request merged**
